universal-application-tool-0.0.1/test/BrowserTest.java

```diff
@@ -31,24 +35,49 @@ protected TestBrowser provideBrowser(int port) {
   }
 
   @Test
-  public void test() {
+  public void homePage() {
     browser.goTo("http://localhost:" + play.api.test.Helpers.testServerPort());
-    assertTrue(browser.pageSource().contains("Your new application is ready."));
+    assertThat(browser.pageSource()).contains("Your new application is ready.");
+  }
+
+  @Test
+  public void applicantProgramList_selectApply_redirectsToEdit() {
+    Program program = insertProgram("My Program");
+    goTo(controllers.applicant.routes.ApplicantProgramsController.index(1L));
+    assertThat(browser.pageSource()).contains("My Program");
+
+    // Redirect when "apply" button is clicked.
+    String applyButtonId = "#apply" + program.id;
+    browser.$(applyButtonId).click();
+    assertThat("/" + browser.url())
+        .isEqualTo(
+            controllers.applicant.routes.ApplicantProgramsController.edit(1L, program.id).url());
   }
 
   @Test
   public void noCredLogin() {
-    String baseUrl = "http://localhost:" + play.api.test.Helpers.testServerPort();
-    browser.goTo(baseUrl + routes.HomeController.loginForm(Optional.empty()).url());
+    goTo(routes.HomeController.loginForm(Optional.empty()));
     browser.$("#guest").click();
     // should be redirected to root.
-    assertEquals("", browser.url());
-    assertTrue(browser.pageSource().contains("Your new application is ready."));
-    browser.goTo(baseUrl + routes.HomeController.secureIndex().url());
-    assertTrue(browser.pageSource().contains("You are logged in."));
-    browser.goTo(baseUrl + routes.ProfileController.myProfile().url());
-    assertTrue(browser.pageSource().contains("GuestClient"));
-    assertTrue(browser.pageSource().contains("{\"created_time\":"));
-    assertTrue(browser.pageSource().contains(Roles.ROLE_APPLICANT.toString()));
+    assertThat(browser.url()).isEmpty();
+    assertThat(browser.pageSource()).contains("Your new application is ready.");
+
+    goTo(routes.HomeController.secureIndex());
+    assertThat(browser.pageSource()).contains("You are logged in.");
+
+    goTo(routes.ProfileController.myProfile());
+    assertThat(browser.pageSource()).contains("GuestClient");
+    assertThat(browser.pageSource()).contains("{\"created_time\":");
+    assertThat(browser.pageSource()).contains(Roles.ROLE_APPLICANT.toString());
+  }
+
+  private void goTo(Call method) {
+    browser.goTo(BASE_URL + method.url());
+  }
```

Comment:
Suggestion (non-blocking): this looks super helpful to have for browser tests! What do you think of adding it to the superclass?

Reply:
Just realized we don't have a superclass for these yet.

I think we'll want to separate our browser tests into multiple files as they get bigger and more involved. If you'd like to split them out now and make a superclass for them go ahead, otherwise please file an issue to do it.

Reply:
Good call - I filed https://github.com/seattle-uat/universal-application-tool/issues/232 and will work on that today